src/libraries/System.Private.CoreLib/src/System/Random.LegacyImpl.cs

```diff
@@ -0,0 +1,223 @@
+                InternalSample() * (1.0 / int.MaxValue);
+
+            public override int Next() => InternalSample();
+
+            public override int Next(int maxValue) => (int)(_parent.Sample() * maxValue);
+
+            public override int Next(int minValue, int maxValue)
+            {
+                long range = (long)maxValue - minValue;
+                return range <= int.MaxValue ?
+                    (int)(_parent.Sample() * range) + minValue :
+                    (int)((long)(GetSampleForLargeRange() * range) + minValue);
+            }
+
+            public override long NextInt64()
+            {
+                while (true)
+                {
+                    // Get top 63 bits to get a value in the range [0, long.MaxValue], but try again
+                    // if the value is actually long.MaxValue, as the method is defined to return a value
+                    // in the range [0, long.MaxValue).
+                    ulong result = NextUInt64() >> 1;
+                    if (result != long.MaxValue)
+                    {
+                        return (long)result;
+                    }
+                }
+            }
+
+            public override long NextInt64(long maxValue) => NextInt64(0, maxValue);
+
+            public override long NextInt64(long minValue, long maxValue)
+            {
+                ulong exclusiveRange = (ulong)(maxValue - minValue);
+
+                if (exclusiveRange > 1)
+                {
+                    // Narrow down to the smallest range [0, 2^bits] that contains maxValue - minValue
+                    // Then repeatedly generate a value in that outer range until we get one within the inner range.
+                    int bits = Log2Ceiling(exclusiveRange);
+                    while (true)
+                    {
+                        ulong result = NextUInt64() >> (sizeof(long) * 8 - bits);
+                        if (result < exclusiveRange)
+                        {
+                            return (long)result + minValue;
+                        }
+                    }
+                }
+
+                Debug.Assert(minValue == maxValue || minValue + 1 == maxValue);
+                return minValue;
+            }
+
+            /// <summary>Produces a value in the range [0, ulong.MaxValue].</summary>
+            private unsafe ulong NextUInt64()
+            {
+                Span<byte> resultBytes = stackalloc byte[8];
+                NextBytes(resultBytes);
+                return BitConverter.ToUInt64(resultBytes);
+            }
+
+            public override double NextDouble() => _parent.Sample();
+
+            public override float NextSingle() => (float)_parent.Sample();
+
+            public override void NextBytes(byte[] buffer)
+            {
+                for (int i = 0; i < buffer.Length; i++)
+                {
+                    buffer[i] = (byte)InternalSample();
```

Comment:
I wonder why this works differently to the span overload.

Reply:
The span overload was added later, at which point it needed to use an existing virtual to ensure that any overrides from a derived type (e.g. locking all access) were taken into account.